### PR TITLE
Add matrix jobs for Java 17 and Kotlin 1.6.0-RC and upgrade Kotlin to 1.5.31 in its job

### PR DESCRIPTION
**Pull request by andrewparmet**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -117,8 +117,11 @@ workflows:
             - build
           matrix:
             parameters:
-              jdk: ["8", "11"]
-              kotlin: ["1.5.0", "1.4.32"]
+              jdk: ["8", "11", "17"]
+              kotlin: ["1.4.32", "1.5.31", "1.6.0-RC"]
+            exclude:
+              - jdk: "17"
```
Comment (andrewparmet) [inline .circleci/config.yml:123]: This combination fails.

File: .circleci/config.yml [hunk 2]
```diff
@@ -117,8 +117,11 @@ workflows:
...
+                kotlin: "1.4.32"
```
Comment (andrewparmet) [inline .circleci/config.yml:124]: This combination fails.

File: .circleci/config.yml (hunk 1 above)
Comment (ogolberg) [inline .circleci/config.yml:123]: don't we need Gradle-7.3-rc to even run on 17?

Comment (andrewparmet) [reply .circleci/config.yml:123]: Sure looks like it ran using 17: https://app.circleci.com/pipelines/github/open-toast/protokt/657/workflows/e215faaf-7e39-4a46-84f9-348eca4e754a/jobs/1084

Comment (ogolberg) [reply .circleci/config.yml:123]: well anyway 7.3 is out

**Review by ogolberg — APPROVED** [review]

**Pull request merged**